### ISSUE TITLE
Log Space for computing Gradients

### ISSUE THREAD

**Issue by pranavsubramani** [pytorch/pytorch#3208]
For certain cases, it may be useful to compute gradients in the log space, is this currently possible to be done within PyTorch? If not, is there a particular place to add this functionality?

**Comment by soumith** [comment]
can you elaborate on what you mean? i cant make sense of your request.

**Comment by pranavsubramani** [comment]
Hmm, in a select few cases, it makes sense in backprop it makes sense to take the logarithm after taking the derivative (for each step in backprop), I was trying to read the docs to see if it's possible to use the .backward function in the log space, but I couldn't find anything. Just wondering if this was possible or not?

**Comment by apaszke** [comment]
I don't think it is possible. This would require us to rewrite all derivative formulas to operate in the log-space. What's your use case? Why do you need log space? Do you have some numerical issues?

**Comment by soumith** [comment]
closing as wontfix, there's no easy way to do this and looks like there's no easy solution other than rewrite all formulas.

**Comment by pranavsubramani** [comment]
Sorry, should have commented earlier, but yeah, there is another way of going about my problem, sorry for not closing it sooner.
